### PR TITLE
entgql: add annotation Deprecated for deprecated field

### PR DESCRIPTION
**Pull request by luantranminh**

### CODE REVIEW

File: entgql/extension.go [hunk 1]
```diff
@@ -244,7 +244,19 @@ func (e *Extension) genSchemaHook() gen.Hook {
...
+				for _, f := range t.DeprecatedFields() {
+					ant, err := annotation(f.Annotations)
+					if err != nil {
+						return err
+					}
+					ant.Directives = append(ant.Directives, Deprecated(f.DeprecationReason()))
+					if f.Annotations == nil {
+						f.Annotations = make(map[string]interface{})
+					}
+					f.Annotations["EntGQL"] = ant
```
Comment (a8m) [inline entgql/extension.go:257]: Make sure the field does not already have such annotation to avoid duplicate directives.

Comment (a8m) [inline entgql/extension.go:257]:
```suggestion
					f.Annotations[ant.Name()] = ant
```

File: entgql/internal/todo/gen_test.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	"path/filepath"
+	"testing"
+
+	"entgo.io/contrib/entgql"
+	"entgo.io/ent/entc"
+	"entgo.io/ent/entc/gen"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGeneratedSchema(t *testing.T) {
```
Comment (a8m) [inline entgql/internal/todo/gen_test.go:28]: Nice, but don't we already test this here: https://github.com/ent/contrib/blob/master/entgql/schema_test.go ?
I don't mind keeping it here.

cc @giautm

Comment (luantranminh) [reply entgql/internal/todo/gen_test.go:28]: schema_test runs tests by calling `buildTypes`. I added logic to `schemaHook`, which executes before `buildTypes`

**Review by giautm — APPROVED** [review]

**Comment by giautm** [comment]
Good Job, welcome a new contributor.

**Pull request merged**